tests/js/validate-function-tests.js

```diff
@@ -0,0 +1,133 @@
+// Unit tests of individual static functions within the validator.
+"use strict";
+var fluid = fluid || require("infusion");
+var gpii  = fluid.registerNamespace("gpii");
+var jqUnit = require("jqUnit");
+
+require("../../src/js/common/validate");
+
+jqUnit.module("Unit tests for validation component static functions...");
+
+jqUnit.test("Test sanitizing basic path segment...", function () {
+    var output = gpii.schema.validator.sanitizePathSegment(".dotless");
+    jqUnit.assertEquals("The output should be as expected...", "dotless", output);
+});
+
+jqUnit.test("Test sanitizing path segment with square brackets...", function () {
+    var output = gpii.schema.validator.sanitizePathSegment(".['this.works']");
+    jqUnit.assertEquals("The output should be as expected...", "this.works", output);
+});
+
+
+jqUnit.test("Extract the path for a missing required top-level field....", function () {
+    var error    = { keyword: "required", dataPath: ".required", message: "message" };
+    var path     = gpii.schema.validator.extractPathSegmentsFromError(error);
```

Comment:
I consolidated all of the tests to use a single runner function and a declarative form.